src/vs/workbench/contrib/scm/browser/dirtyDiffSwitcher.ts

```diff
@@ -0,0 +1,60 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as nls from 'vs/nls';
+import { Action, IAction } from 'vs/base/common/actions';
+import { IContextViewService } from 'vs/platform/contextview/browser/contextView';
+import { ISelectOptionItem } from 'vs/base/browser/ui/selectBox/selectBox';
+import { SelectActionViewItem } from 'vs/base/browser/ui/actionbar/actionViewItems';
+import { defaultSelectBoxStyles } from 'vs/platform/theme/browser/defaultStyles';
+import { IThemeService } from 'vs/platform/theme/common/themeService';
+import { peekViewTitleBackground } from 'vs/editor/contrib/peekView/browser/peekView';
+import { Color } from 'vs/base/common/color';
+
+interface IQuickDiffSelectItem extends ISelectOptionItem {
+	provider: string;
+}
+
+export class SwitchQuickDiffViewItem extends SelectActionViewItem {
+	private readonly optionsItems: IQuickDiffSelectItem[];
+
+	constructor(
+		action: IAction,
+		providers: string[],
+		selected: string,
+		@IContextViewService contextViewService: IContextViewService,
+		@IThemeService themeService: IThemeService
+	) {
+		const items = providers.map(provider => ({ provider, text: provider }));
+		let startingSelection = providers.indexOf(selected);
+		if (startingSelection === -1) {
+			startingSelection = 0;
+		}
+		const styles = { ...defaultSelectBoxStyles };
+		const theme = themeService.getColorTheme();
+		styles.selectBackground = theme.getColor(peekViewTitleBackground)?.lighten(.7).toString() || Color.transparent.toString();
+		super(null, action, items, startingSelection, contextViewService, styles, { ariaLabel: nls.localize('remotes', 'Switch quick diff base') });
+		this.optionsItems = items;
+	}
+
+	public setSelection(provider: string) {
+		const index = this.optionsItems.findIndex(item => item.provider === provider);
+		this.select(index);
+	}
+
+	protected override getActionContext(_: string, index: number): any {
```

Comment:
Should this be better typed?

```suggestion
	protected override getActionContext(_: string, index: number): IQuickDiffSelectItem {
```

If unsupported, I suggest to make things better by adding generics to `SelectActionViewItem`.